### PR TITLE
add documentation for new microsoft auth resolver

### PR DESCRIPTION
**Pull request by JessicaJHee**
## Description

This PR is a follow up to add documentation for this [PR](https://github.com/backstage/backstage/pull/26153). 

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

cc: @awanlin

### CODE REVIEW

File: docs/auth/microsoft/provider.md [hunk 1]
```diff
@@ -93,6 +94,7 @@ This provider includes several resolvers out of the box that you can use:
 - `emailMatchingUserEntityProfileEmail`: Matches the email address from the auth provider with the User entity that has a matching `spec.profile.email`. If no match is found it will throw a `NotFoundError`.
 - `emailLocalPartMatchingUserEntityName`: Matches the [local part](https://en.wikipedia.org/wiki/Email_address#Local-part) of the email address from the auth provider with the User entity that has a matching `name`. If no match is found it will throw a `NotFoundError`.
 - `emailMatchingUserEntityAnnotation`: Matches the email address from the auth provider with the User entity where the value of the `microsoft.com/email` annotation matches. If no match is found it will throw a `NotFoundError`.
+- `userIdMatchingUserEntityAnnotation`: Matches the user profile ID from the auth provider with the User entity where the value of the `graph.microsoft.com/user-id` annotation matches. If no match is found it will throw a `NotFoundError`.
```
Comment (awanlin) [inline docs/auth/microsoft/provider.md:97]: Should we add a note about this being recommended for those who don't have email?

**Review by awanlin — CHANGES_REQUESTED** [review]
Thanks for this @JessicaJHee! 

I actually think this works out well being in it's own PR. As we don't have versioned documentation yet let get this PR in a good state and then merge it after the next release that will take place in late September 👍

File: docs/auth/microsoft/provider.md (hunk 1 above)
Comment (JessicaJHee) [reply docs/auth/microsoft/provider.md:97]: Yes! Sorry I missed this review comment - will add this now.

**Review by jhaals — APPROVED** [review]
Looking good 👍

**Review by awanlin — COMMENTED** [review]
Thanks again for this @JessicaJHee 🚀 

Just a heads up we added versioning to the docs shortly after this PR was created so now this change will only show under the "next" option in the version picker, it will part of "stable" after the October release.

**Pull request merged**